lesson2/homework.js

```diff
@@ -0,0 +1,30 @@
+
+
+const addNewTask = function(id, title, description, created, updated, dueDate, statusID, userID) {
+  
+};
+
+const changeTaskTitle = function(taskID, newTitle) {
+  
+};
+
+const changeTaskDueDate = function(taskID, newDueDate) {
+  
+};
+
+const changeTaskStatus = function(taskID, newStatus) {
+  
+};
+
+const markTaskAsCompleted = function(taskID, newStatus) {
+  
+};
+
+const deleteTask = function(taskID, taskID) {
+  
+};
+
+const deleteUser = function(taskID, userID) {
+  
+};
+
```

Comment:
Down here we could put the validating code that runs all the functions one by one. Fx:

```
// Add a new task
const newTask = addNewTask('Wash dishes', 'blah blah', ....);
// Verify that it got an ID
if (newTask.id < 1) {
    console.log('Expected newly added task to have an ID');
    process.exit(1);
}

// Find task by ID
const foundTask = findTaskByID(newTask.id);
if (!foundTask) {
    console.log('Could not find task by ID');
    process.exit(1);
}

// Change task title
const newTitle = 'This is a new task title';
changeTaskTitle(newTask.id, newTitle);

const updatedTask = findTaskByID(newTask.id);
if (updatedTask.title !== newTitle) {
    console.log('Expected task to have new title');
    process.exit(1);    
}

```